engine/servermaster/server.go

```diff
@@ -814,21 +801,44 @@ func (s *Server) runLeaderService(ctx context.Context) (err error) {
 		s.executorManager,
 		s.resourceManagerService)
 
-	s.gcRunner = externRescManager.NewGCRunner(s.frameMetaClient, executorClients, &s.cfg.Storage)
-	s.gcCoordinator = externRescManager.NewGCCoordinator(s.executorManager, s.jobManager, s.frameMetaClient, s.gcRunner)
-
 	// TODO refactor this method to make it more readable and maintainable.
 	errg, errgCtx := errgroup.WithContext(ctx)
 
 	errg.Go(func() error {
-		return s.gcRunner.Run(errgCtx)
+		return s.executorManager.Run(errgCtx)
 	})
+
 	errg.Go(func() error {
-		return s.gcCoordinator.Run(errgCtx)
+		updater := executorInfoUpdater{
+			msgRouter:     s.p2pMsgRouter,
+			executorGroup: executorClients,
+		}
+		return serverutil.WatchExecutors(errgCtx, s.executorManager, updater)
 	})
 
+	s.leaderDegrader.updateExecutorManager(true)
```

Comment:
Whether this field should be updated inside `s.executorManager.Run`？

Reply:
Not necessary, `executorManager` can serve after it is created. API of `executorManager` can run concurrently against logic in `Run`